.github/workflows/build.yml

```diff
@@ -82,8 +78,22 @@ jobs:
           subscription-id: ${{ env.AZURE_SUBSCRIPTION_ID }}
         if: ${{ env.AZURE_CLIENT_ID != '' && env.AZURE_TENANT_ID != '' && env.AZURE_SUBSCRIPTION_ID != '' }}
 
-      - name: Build with Maven (Native) and run tests against Azure services
-        run: .github/build-with-maven-native.sh
+      - name: Create Azure resources
+        run: .github/create-azure-resources.sh
+        if: ${{ env.AZURE_CLIENT_ID != '' && env.AZURE_TENANT_ID != '' && env.AZURE_SUBSCRIPTION_ID != '' }}
+
+      - name: Az CLI login
```

Comment:
We add `  - name: Az CLI login
        uses: azure/login@v1` several times in the workflow, and I know we do it for a reason, but I forget about it? Could you explain it for me?

Reply:
Sure. During the development and test phase, I observed executing integration test for `azure-cosmos` extension failed. After investigation, it's due to building native executables spends too much time which caused the token expire when test is executed. The solution is to separate into sub-steps and log into azure before each step to ensure azure login is active.